pkg/kademlia/dialer.go

```diff
@@ -88,6 +88,23 @@ func (dialer *Dialer) Ping(ctx context.Context, target pb.Node) (bool, error) {
 	return err == nil, errs.Combine(err, conn.disconnect())
 }
 
+// PingAddress pings target by address (no node ID verification).
+func (dialer *Dialer) PingAddress(ctx context.Context, address string) (bool, storj.NodeID, error) {
+	if !dialer.limit.Lock() {
+		return false, storj.NodeID{}, context.Canceled
+	}
+	defer dialer.limit.Unlock()
+
+	conn, err := dialer.dialAddress(ctx, address)
+	if err != nil {
+		return false, storj.NodeID{}, err
+	}
+
+	res, err := conn.client.Ping(ctx, &pb.PingRequest{})
```

Comment:
We cannot trust the ID response, we must use the id that was communicated during the tls handshake.